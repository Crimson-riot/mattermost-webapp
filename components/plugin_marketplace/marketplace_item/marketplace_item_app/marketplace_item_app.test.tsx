// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import MarketplaceItemPlugin, {MarketplaceItemAppProps} from './marketplace_item_app';

describe('components/MarketplaceItemPlugin', () => {
    describe('MarketplaceItem', () => {
        const baseProps: MarketplaceItemAppProps = {
            id: 'id',
            name: 'name',
            description: 'test plugin',
            homepage_url: 'http://example.com',
            root_url: 'http://example.com/install',
            installed: false,
            trackEvent: jest.fn(() => {}),
            actions: {
                installApp: jest.fn(() => {}),
                closeMarketplaceModal: jest.fn(() => {}),
            },
        };

        test('should render', () => {
            const wrapper = shallow<MarketplaceItemPlugin>(
                <MarketplaceItemPlugin {...baseProps}/>,
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should render with no plugin description', () => {
            const props = {...baseProps};
            delete props.description;

            const wrapper = shallow(
                <MarketplaceItemPlugin {...props}/>,
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should render with no homepage url', () => {
            const props = {...baseProps};
            delete props.homepage_url;

            const wrapper = shallow<MarketplaceItemPlugin>(
                <MarketplaceItemPlugin {...props}/>,
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should render installed app', () => {
            const props = {
                ...baseProps,
                installed: true,
            };

            const wrapper = shallow<MarketplaceItemPlugin>(
                <MarketplaceItemPlugin {...props}/>,
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should render with empty list of labels', () => {
            const props = {
                ...baseProps,
                labels: [],
            };

            const wrapper = shallow<MarketplaceItemPlugin>(
                <MarketplaceItemPlugin {...props}/>,
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should render with one labels', () => {
            const props = {
                ...baseProps,
                labels: [
                    {
                        name: 'someName',
                        description: 'some description',
                        url: 'http://example.com/info',
                    },
                ],
            };

            const wrapper = shallow<MarketplaceItemPlugin>(
                <MarketplaceItemPlugin {...props}/>,
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should render with two labels', () => {
            const props = {
                ...baseProps,
                labels: [
                    {
                        name: 'someName',
                        description: 'some description',
                        url: 'http://example.com/info',
                    }, {
                        name: 'someName2',
                        description: 'some description2',
                        url: 'http://example.com/info2',
                    },
                ],
            };

            const wrapper = shallow<MarketplaceItemPlugin>(
                <MarketplaceItemPlugin {...props}/>,
            );

            expect(wrapper).toMatchSnapshot();
        });

        describe('install should trigger track event and close modal', () => {
            const props = {
                ...baseProps,
                isDefaultMarketplace: true,
            };

            const wrapper = shallow<MarketplaceItemPlugin>(
                <MarketplaceItemPlugin {...props}/>,
            );

            wrapper.instance().onInstall();
            expect(props.trackEvent).toBeCalledWith('plugins', 'ui_marketplace_install_app', {
                app_id: 'id',
            });
            expect(props.actions.installApp).toHaveBeenCalledWith('id', 'http://example.com/install');
            expect(props.actions.closeMarketplaceModal).toBeCalled();
        });
    });
});
