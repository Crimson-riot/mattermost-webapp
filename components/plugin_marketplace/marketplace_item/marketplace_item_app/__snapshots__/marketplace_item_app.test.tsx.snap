// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MarketplaceItemPlugin MarketplaceItem should render 1`] = `
<Fragment>
  <MarketplaceItem
    actions={
      Object {
        "closeMarketplaceModal": [MockFunction],
        "installApp": [MockFunction],
      }
    }
    button={
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Install"
          id="marketplace_modal.list.Install"
        />
      </button>
    }
    description="test plugin"
    homepage_url="http://example.com"
    id="id"
    installed={false}
    name="name"
    root_url="http://example.com/install"
    trackEvent={[MockFunction]}
    updateDetails={null}
    versionLabel={null}
  />
</Fragment>
`;

exports[`components/MarketplaceItemPlugin MarketplaceItem should render installed app 1`] = `
<Fragment>
  <MarketplaceItem
    actions={
      Object {
        "closeMarketplaceModal": [MockFunction],
        "installApp": [MockFunction],
      }
    }
    button={
      <button
        className="btn btn-outline"
        disabled={true}
      >
        <FormattedMessage
          defaultMessage="Installed"
          id="marketplace_modal.list.installed"
        />
      </button>
    }
    description="test plugin"
    homepage_url="http://example.com"
    id="id"
    installed={true}
    name="name"
    root_url="http://example.com/install"
    trackEvent={[MockFunction]}
    updateDetails={null}
    versionLabel={null}
  />
</Fragment>
`;

exports[`components/MarketplaceItemPlugin MarketplaceItem should render with empty list of labels 1`] = `
<Fragment>
  <MarketplaceItem
    actions={
      Object {
        "closeMarketplaceModal": [MockFunction],
        "installApp": [MockFunction],
      }
    }
    button={
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Install"
          id="marketplace_modal.list.Install"
        />
      </button>
    }
    description="test plugin"
    homepage_url="http://example.com"
    id="id"
    installed={false}
    labels={Array []}
    name="name"
    root_url="http://example.com/install"
    trackEvent={[MockFunction]}
    updateDetails={null}
    versionLabel={null}
  />
</Fragment>
`;

exports[`components/MarketplaceItemPlugin MarketplaceItem should render with no homepage url 1`] = `
<Fragment>
  <MarketplaceItem
    actions={
      Object {
        "closeMarketplaceModal": [MockFunction],
        "installApp": [MockFunction],
      }
    }
    button={
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Install"
          id="marketplace_modal.list.Install"
        />
      </button>
    }
    description="test plugin"
    id="id"
    installed={false}
    name="name"
    root_url="http://example.com/install"
    trackEvent={[MockFunction]}
    updateDetails={null}
    versionLabel={null}
  />
</Fragment>
`;

exports[`components/MarketplaceItemPlugin MarketplaceItem should render with no plugin description 1`] = `
<Fragment>
  <MarketplaceItem
    actions={
      Object {
        "closeMarketplaceModal": [MockFunction],
        "installApp": [MockFunction],
      }
    }
    button={
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Install"
          id="marketplace_modal.list.Install"
        />
      </button>
    }
    homepage_url="http://example.com"
    id="id"
    installed={false}
    name="name"
    root_url="http://example.com/install"
    trackEvent={[MockFunction]}
    updateDetails={null}
    versionLabel={null}
  />
</Fragment>
`;

exports[`components/MarketplaceItemPlugin MarketplaceItem should render with one labels 1`] = `
<Fragment>
  <MarketplaceItem
    actions={
      Object {
        "closeMarketplaceModal": [MockFunction],
        "installApp": [MockFunction],
      }
    }
    button={
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Install"
          id="marketplace_modal.list.Install"
        />
      </button>
    }
    description="test plugin"
    homepage_url="http://example.com"
    id="id"
    installed={false}
    labels={
      Array [
        Object {
          "description": "some description",
          "name": "someName",
          "url": "http://example.com/info",
        },
      ]
    }
    name="name"
    root_url="http://example.com/install"
    trackEvent={[MockFunction]}
    updateDetails={null}
    versionLabel={null}
  />
</Fragment>
`;

exports[`components/MarketplaceItemPlugin MarketplaceItem should render with two labels 1`] = `
<Fragment>
  <MarketplaceItem
    actions={
      Object {
        "closeMarketplaceModal": [MockFunction],
        "installApp": [MockFunction],
      }
    }
    button={
      <button
        className="btn btn-primary"
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Install"
          id="marketplace_modal.list.Install"
        />
      </button>
    }
    description="test plugin"
    homepage_url="http://example.com"
    id="id"
    installed={false}
    labels={
      Array [
        Object {
          "description": "some description",
          "name": "someName",
          "url": "http://example.com/info",
        },
        Object {
          "description": "some description2",
          "name": "someName2",
          "url": "http://example.com/info2",
        },
      ]
    }
    name="name"
    root_url="http://example.com/install"
    trackEvent={[MockFunction]}
    updateDetails={null}
    versionLabel={null}
  />
</Fragment>
`;
